library_management_cookiecutter/library/views.py

```diff
@@ -0,0 +1,243 @@
+                assignment.user = user
+                assignment.save()
+
+                total_books_assign = AssignedBook.objects.filter(book = book,is_deleted = False).count()
+                rem = book.quantity - total_books_assign
+                return JsonResponse({"message":"successfully book assign.","rem":rem,"book_id":book.id})
+
+class AssignBookUser(View):
+    template_name = 'user_assign_book_list.html'
+    model = AssignedBook
+    queryset = AssignedBook.objects.filter(is_deleted = False)
+
+    def get(self, request,*args, **kwargs):
+        assign_book_list = list(AssignedBook.objects.filter(user= request.user,is_deleted=False))
+        assign_book_data = [{'id': assign_book.book.id, 'book': assign_book.book.book_name} for assign_book in assign_book_list]
+        return render(self.request,self.template_name,context = {"assign_book_data":assign_book_data})
+
+    def post(self, request, *args, **kwargs):
+        user = request.user
+        book = Book.objects.get(id=request.POST.get("book"))
+        
+        if AssignedBook.objects.filter(book = book,user= user,is_deleted = False).exists():
+            if request.POST.get('button_action') == "return_book":
+
+                assigned_book = AssignedBook.objects.get(book = book,user= user,is_deleted = False)
+                assigned_book.is_deleted = True
+                assigned_book.date_returned = timezone.now()
+                assigned_book.save()
+
+                response = {
+                "status": True,
+                "message": "Book Returened successfully",
+                }
+                return JsonResponse(response,safe=False)
+            else:
+                data_list=[]
+                for books in AssignedBook.objects.filter(book = book,user=user):
+                    book_list = {
+                        "date_borrowed": books.date_borrowed,
+                        "date_return":books.date_returned,
+                    }
+                    data_list.append(book_list)
+                return JsonResponse(data_list,safe=False)
+        else:
+            return JsonResponse({"message":"done"})
+
+
+class BookHistory(TemplateView):
+    template_name = "book_history.html"
+    model =AssignedBook
+
+    def post(self,request,*args,**kwargs):
+        assigned_books = AssignedBook.objects.values("book").annotate(count=Count("book"))
+        book_list = []
+
+        for assigned_book in assigned_books:
+            book = Book.objects.get(id=assigned_book['book'])
+            assign_username =  AssignedBook.objects.filter(book=book, is_deleted=False).values_list('user__username', flat=True)
+            assignments_count = assign_username.count()
+            # assign_username = assign_book
+            return_name = AssignedBook.objects.filter(book=book, is_deleted=True).values_list('user__username',flat=True)
+            returns_count = return_name.count()
+            # return_name = returned_count
```

Comment:
remove comments. if you think comment should be helpful then keep it otherwise use print only for reference.